src/Raven.Server/Documents/AttachmentsStorage.cs

```diff
@@ -901,6 +901,64 @@ public AttachmentDetails MoveAttachment(DocumentsOperationContext context, strin
             return result;
         }
 
+        public AttachmentDetails RenameAttachment(DocumentsOperationContext context, Slice lowerId, string name, string hash, string contentType, AttachmentType attachmentType, string newName)
```

Comment:
Why is this required? The tag on the attachment is for debug only.
We don't actually need to rename it.

Reply:
yes the unique value is key, but we don't allow attachments to have same name when it is added from studio or Client API. Conflicts in replication can cause such scenario, which in place can break other things like patch.

Reply:
What happens if we have _another_ conflict? Then we'll have a duplicate on `resolved_name`, no?

Reply:
done